scipy/stats/tests/test_distributions.py

```diff
@@ -6948,6 +6948,26 @@ def optimizer(func, x0, args=(), disp=0):
     assert_allclose(scale, 3.099456, rtol=5e-6)
 
 
+# Expected values were computed with mpmath.
+@pytest.mark.parametrize('x, c, expected',
+                         [(3, 1.5, 0.175064510070713299327),
+                          (2000, 1.5, 1.11802773877318715787e-5),
+                          (2000, 9.25, 2.92060308832269637092e-31),
+                          (1e15, 1.5, 3.16227766016837933199884e-23)])
+def test_invweibull_sf(x, c, expected):
+    computed = stats.invweibull.sf(x, c)
+    assert_allclose(computed, expected, rtol=1e-15)
+
+
+# Expected values were computed with mpmath.
+@pytest.mark.parametrize('p, c, expected',
+                         [(0.5, 2.5, 1.15789669836468183976),
+                          (3e-18, 5, 3195.77171838060906447)])
```

Comment:
```suggestion
                          (3e-18, 5, 3195.77171838060902814)])
```
Not that it matters, but
```python3
from mpmath import log1p, mpf, mp
mp.dps = 100
p, c = mpf('3e-18'), mpf('5')
print((-log1p(-p))**(-1/c))
# 3195.7717183806090**28135**108441581961556067026704139393111927235801060392214378263341280284497431151393
```

Reply:
Actually, the expected value should be the best value that can be expected, *given that the input is 64 bit floating point*.  So the `mpf` versions of `p` and `c` should be created from the same 64 floating point (or integer values) that would actually be passed to the `isf` method.  In this case, that means the inputs to `mpf` should not be passed in as strings, since `mpf('3e-18')` ≠ `mpf(3e-18)`:

```
In [8]: from mpmath import log1p, mpf, mp
   ...: mp.dps = 100
   ...: p, c = mpf(3e-18), mpf(5)
   ...: print((-log1p(-p))**(-1/c))
3195.771718380609064472889988018864363815582980215406228722937454943560087624947042415392958803589555
```

Reply:
Oops, I used to do that. I was thinking of the Wolfram Alpha trick to get more precise output.